files/rsyslog/rsyslog.sh

```diff
@@ -7,25 +7,35 @@ export RSYSLOG_WORKDIRECTORY=${RSYSLOG_WORKDIRECTORY:-/var/lib/rsyslog.pod}
 export MERGE_JSON_LOG=${MERGE_JSON_LOG:-false}
 export UNDEFINED_DEBUG=${UNDEFINED_DEBUG:-false}
 # if false, mmexternal undefined_field won't be called.
-export USE_MMEXTERNAL=${USE_MMEXTERNAL:-true}
 # if true, mmnormalize skipempty is executed.
 # effective when USE_MMEXTERNAL is false.
-export SKIP_EMPTY=${SKIP_EMPTY:-false}
 export RSYSLOG_FILE_READ_FROM_TAIL=${RSYSLOG_FILE_READ_FROM_TAIL:-off}
 export RSYSLOG_JOURNAL_READ_FROM_TAIL=${RSYSLOG_JOURNAL_READ_FROM_TAIL:-off}
 
-# Create a config file for undefined fields from env vars
+CDM_USE_UNDEFINED=${CDM_USE_UNDEFINED:-false}
+CDM_KEEP_EMPTY_FIELDS=${CDM_KEEP_EMPTY_FIELDS:-""}
+CDM_UNDEFINED_TO_STRING=${CDM_UNDEFINED_TO_STRING:-false}
+CDM_UNDEFINED_DOT_REPLACE_CHAR=${CDM_UNDEFINED_DOT_REPLACE_CHAR:-"UNUSED"}
 undefined_conf="/var/lib/rsyslog.pod/undefined.json"
-echo "{\"CDM_USE_UNDEFINED\" : ${CDM_USE_UNDEFINED:-false}, \
+if [ "${CDM_USE_UNDEFINED}" = "false" -a "${CDM_KEEP_EMPTY_FIELDS}" = "" -a "${CDM_UNDEFINED_TO_STRING}" = "false" -a "${CDM_UNDEFINED_DOT_REPLACE_CHAR}" = "UNUSED" ] ; then
+    export SKIP_EMPTY=true
+    export USE_MMEXTERNAL=false
+    rm -f $undefined_conf
+else
+    export SKIP_EMPTY=false
+    export USE_MMEXTERNAL=true
+    # Create a config file for undefined fields from env vars
```

Comment:
```bash
if [ "$SKIP_EMPTY" = false -a "$USE_MMEXTERNAL" = true ] ; then
  create $undefined_conf
else
  rm -f $undefined_conf
fi
```

Reply:
Thanks!  I'm going to modify this PR as suggested...

Reply:
Yes - buildcop status is Severe - because we're all trying to merge 14,000,012 PRs by tomorrow :-(